### ISSUE TITLE
Example application fails with FIDO_ERR_INTERNAL (-9)

### ISSUE THREAD

**Issue by marsupilami79** [mikerabat/DelphiFido2#3]
Hello,

I am not sure what is the right way to ask for help. I was doing some testing with the example appication. But whatever I do, creating credentials or getting an assertion fails with FIDO_ERR_INTERNAL (-9).
I tested this with a Yubikey 5 (Firmware 5.4.3) using WIndows Hello and direct access.

When trying to create credentials, I get:
`Failed: fido_dev_make_cred: FIDO_ERR_INTERNAL (-9)`

And when creating an assertion, I get:
`Failed: fido_dev_get_assert: FIDO_ERR_INTERNAL (-9)`

Environment:
  * Windows 10 / 11
  * Delphi 11.3
  * libfido2 1.13.0
  * everything compiled for WIN32

Do you have any suggestions?

**Comment by mikerabat** [comment]
Before I dive into that... Did you start the application with Admin Priviledges (UAC)?

**Comment by marsupilami79** [comment]
Yes I started the applilcation with Admin Privileges.

I did some more digging. On my computer the Yubikey is listed first and Windows hello is listed as the second.
Since fUSBPath is only set in btnCheckKeyClick and there it gets set to the last detected device, Windows Hello will always be used on my computer. The choice I make in cboDevs doesn't matter.

The Yubikey works as expected when explicitly setting fUSBPath to my Yubikey in btnCreateCredClick:
`fUSBPath := '\\?\hid#vid_1050&pid_0407&mi_01#7&2a95a7b2&0&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}';`

I could make a patch that stores the USB paths of all detected devices and sets fUSBPath to the desired device when cboDevs changes.

So FIDO_ERR_INTERNAL will only happen with Windows Hello.

**Comment by mikerabat** [comment]
Yes the left side of the test program, wich only uses the raw dll api, simply uses the last found key... 
the right one, which uses the object wrapper, use the combo box... that can be "optimized"...

I guess I need to check back with the dll vendor Yubikey why that actually is - I can reproduce the problem here too.

**Comment by mikerabat** [comment]
I finally got back some info from the developers at yubiko. 

My response:

> > Basically the conclusion is that a "windows hello" key can only be
> > used (in parts) if they selected a fido key in the first place.
> > I configured my Windows with a fingerprint touch sensor which I think
> > is the "Win hello" device... didn't work with that though I always
> > got an internal error back.
> 

His reponse:
> That's probably it -- the underlying TPM used by Windows Hello to mimic a FIDO2 device implements only a subset of the functionality of a Yubikey. You are probably using a combination of capabilities (algorithm, FIDO2 credential options) that it does not support.


Microsoft started to their own (a bit limited) library in the WinHello world: 
Webauth.dll 
It's more limited than the Fido2.dll but the api also used in webbrowsers. The project contains a conversition from the
header Microsoft provided and a small demo that shows how to use the api (it's a bit older so it's based on V2 - the current
version is V7).
